spec/mongo/index/view_spec.rb

```diff
@@ -196,7 +196,70 @@
         expect(view.get('random_name')).to_not be_nil
       end
     end
-  end
+
+    context 'when providing an invalid partial index filter' do
+
+      let(:spec) do
+        { random:1 }
+      end
+
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: 5)
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {"x": {"$asda": 3}})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {
+                                  "$and": [{"$and": [{"x": {"$lt": 2}},
+                                                     {"x": {"$gt": 0}}]},
+                                           {"x": {"$exists": true}}]})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+    end
+
+    context 'when providing a valid partial index filter' do
+      let!(:result) do
+        view.create_one({"x": 1}, partial_filter_expression: {
+                                    "a": {"$lte": 1.5}})
```

Comment:
The ! of the let is not necessary. The difference between defining result as 
```ruby
let(:result)
```
and 
```ruby
let!(:result)
```

is that the first is lazily evaluated. In other words, it will be evaluated when you refer to it in the actually test (i.e. the 'it' block)
When you use a '!', the result is evaluated before the 'it' block runs. Sometimes this is necessary, but not in this case.
So you can just use lazy-evaluation and define the result as 

```ruby
let(:result)
```